airflow/cli/commands/task_command.py

```diff
@@ -75,9 +75,9 @@ def _generate_temporary_run_id() -> str:
 def _get_dag_run(
     *,
     dag: DAG,
-    exec_date_or_run_id: str,
     create_if_necessary: CreateIfNecessary,
-    session: Session,
+    exec_date_or_run_id: Optional[str] = None,
+    session: Session = NEW_SESSION,
```

Comment:
If we make this default session change, should we also `@provide_session` this?

Reply:
i don't see why the one should prompt the other.  the NEW_SESSION change doesn't really change the behavior of the function. the reason it's there in general is this:

```
# A fake session to use in functions decorated by provide_session. This allows
# the 'session' argument to be of type Session instead of Optional[Session],
# making it easier to type hint the function body without dealing with the None
# case that can never happen at runtime.
```

but for me, i'm adding it here simply because i want to keep it as the last argument, so to do so, it needs a default, since i'm making exec date optional.

but what do you think?

Reply:
All arguments in this function are keyword-only, so ordering does not matter. You can add the new argument anywhere (but behind `*`) and keep all other arguments as-is.

Reply:
@ashb has previously requested that we keep session as the last argument 🤷 

@ashb ?

Reply:
You can keep it last, keyword-only means _you don’t need to add a default_ to keep it last.

Reply:
🤯